src/core/initializeComponentsFactory.js

```diff
@@ -10,20 +10,31 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-import createLifecycle from "./createLifecycle";
-import createComponentRegistry from "./createComponentRegistry";
-import createNetwork from "./network";
 import { stackError } from "../utils";
-import createOptIn from "./createOptIn";
+import cookieDetails from "../constants/cookieDetails";
+
+const { ALLOY_COOKIE_NAME, ALLOY_COOKIE_EXPIRES } = cookieDetails;
 
 export default (
   componentCreators,
   logger,
   createNamespacedStorage,
-  cookie
+  createCookieProxy,
+  createCookie,
+  createLifecycle,
+  createComponentRegistry,
+  createNetwork,
+  createOptIn
 ) => config => {
   const componentRegistry = createComponentRegistry();
   const { imsOrgId, propertyId, cookieDomain } = config;
+  const cookieName = `${ALLOY_COOKIE_NAME}_${propertyId}`;
+  const cookieProxy = createCookieProxy(
```

Comment:
If we ever support multiple instances with the same propertyId, we could memoize the createCookieProxy method so that if two cookies are fetched with the same name and domain, it would actually return the same instance.